mifosng-provider/src/main/java/org/mifosplatform/portfolio/group/service/GroupingTypesWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -787,11 +805,20 @@ public CommandProcessingResult associateGroupsToCenter(final Long centerId, fina
     public CommandProcessingResult disassociateGroupsToCenter(final Long centerId, final JsonCommand command) {
         this.fromApiJsonDeserializer.validateForDisassociateGroups(command.json());
 
+        Group groupForUpdate = null;
         final Group centerForUpdate = this.groupRepository.findOneWithNotFoundDetection(centerId);
         final Set<Group> groupMembers = assembleSetOfChildGroups(centerForUpdate.officeId(), command);
 
         final Map<String, Object> actualChanges = new HashMap<>();
 
+		for (final Group group : groupMembers) {
+			Long groupId = group.getId();
+			groupForUpdate = this.groupRepository
+					.findOneWithNotFoundDetection(groupId);
+			groupForUpdate
+					.generateHierarchyToDisassociateGroups(groupForUpdate);
+		}
```

Comment:
this is confusing.

You have a method call on an object groupForUpdate to which you are passing groupForUpdate , the object itself) as a parameter


You should ideally have had a method like resetGroupHierarchy(); which accpts no parameters and resets the group hierarchy for a group with no parents